src/notification/index.js

```diff
@@ -233,8 +228,24 @@ export class NotificationListener {
 
   /** Start listening.  Don't call twice without intervening `stop`. */
   start() {
-    this.listen('notificationOpened', this.handleNotificationOpen);
+    if (Platform.OS === 'android') {
+      // On Android, the object passed to the handler is constructed in
+      // FcmMessage.kt, and will always be a Notification.
+      this.listen('notificationOpened', this.handleNotificationOpen);
+    } else {
+      // On iOS, `note` should be an IOSNotifications object. The notification
+      // data it returns from `getData` is unvalidated -- it comes almost
+      // straight off the wire from the server.
+      this.listen('notificationOpened', (note: { getData(): JSONableDict }) => {
+        const data = verifyIosNotificationData(note);
```

Comment:
Hmm, something doesn't seem right in these types! We're passing `verifyIosNotificationData` the thing that has a `getData()` method, rather than that method's return value.

I'm actually puzzled here that Flow doesn't give an error.

Reply:
> I'm actually puzzled here that Flow doesn't give an error.

This is, alas, a known bug. See comment in `jsonable.js`.

Reply:
Ah, I see. It's because `JSONableDict`, which is the type of that parameter, is an inexact object type. (Which in turn it has to be because of that Flow bug fixed in v0.111.)